Date.ts

```diff
@@ -1,17 +1,19 @@
+import { isly } from "isly"
 import { DateSpan } from "./DateSpan"
 import { Locale } from "./Locale"
 
 export type Date = string
 
 export namespace Date {
-	export function is(value: any | Date): value is Date {
-		return (
-			typeof value == "string" &&
-			value.length == 10 &&
-			new globalThis.Date(value).toString() != "Invalid Date" &&
-			create(new globalThis.Date(value)) == value
+	export const type = isly.named(
+		"isoly.Date",
+		isly.string<Date>( // Should work for all leap years from 1800 to 2499
+			/^(((18|19|20|21|22|23|24)[0-9]{2}-(0[13578]|1[02])-(0[1-9]|[12][0-9]|3[01]))|(18|19|20|21|22|23|24)[0-9]{2}-(0[469]|11)-(0[1-9]|[12][0-9]|30)|(18|19|20|21|22|23|24)[0-9]{2}-(02)-(0[1-9]|1[0-9]|2[0-8])|(((18|19|20|21|22|23|24)(04|08|[2468][048]|[13579][26]))|2000|2400)-(02)-29)$/
```

Comment:
Instead of checking this confusingly long regex, maybe check
length == 10
1<= month <=12
``` ts
If febuary
  const isLeapYear = (year % 4 === 0 && (year % 100 !== 0 || year % 400 === 0));
  daysOfTheMonth = isLeapYear ? 29 : 28;
else 
   daysOfTheMonth = ...
```

0 <= day <= daysOfTheMonth

Reply:
So you suggest updating isly to support a function for checking the type?

Reply:
No, just use the `isly.isFrom` or whatever it's valled.

Reply:
```ts
export const type = isly.fromIs<Date>("isoly.Date", (value: any): value is Date => {
	let result: boolean
	if (typeof value == "string") {
		const followPattern = /^\d{4}-\d{2}-\d{2}$/.test(value)
		const month = getMonth(value)
		const validMonth = month >= 1 && month <= 12
		const maxDays = daysInMonth(value)
		const day = getDay(value)
		const validDay = day >= 1 && day <= maxDays
		result = followPattern && validMonth && validDay
	} else {
		result = false
	}
	return result
})

	
	
export function daysInMonth(date: Date): number {
	let result: number
	const year = getYear(date)
	const month = getMonth(date)
	if (month == 2) {
		const isLeapYear = year % 4 === 0 && (year % 100 !== 0 || year % 400 === 0)
		result = isLeapYear ? 29 : 28
	} else {
		result = { 1: 31, 3: 31, 4: 30, 5: 31, 6: 30, 7: 31, 8: 31, 9: 30, 10: 31, 11: 30, 12: 31 }[month] as number
	}
	return result
}
```

Reply:
Modified how DateTime and Date are checked.